mesh/nurbs.hpp

```diff
@@ -65,6 +65,11 @@ class KnotVector
        order @a order and number of control points @a NCP. */
    KnotVector(int order, int NCP);
 
+   /** @brief Create a KnotVector by passing in a degree, a Vector of interval
+       lengths of length n, and a list of continuity of length n + 1 */
```

Comment:
This condition `continuity.Size() == intervals.Size() + 1` is not verified in the function. Should it be? Also, some more documentation (after the `@brief`) would be helpful, to further explain `intervals` and `continuity`. In particular, the range of possible values for `continuity` entries and their meaning would be useful information.

Reply:
I added a check for this condition in the latest version, plus some more details in the documentation comment on the interpretation of `intervals` and `continuity`.  

One condition that's *not* checked yet is that the continuities are within the allowed bounds.  However, I favored leaving out an assert that has O(N) cost to check.  (I wouldn't anticipate this constructor being in performance critical places, though.  Should I maybe include it within a debug `#ifdef` / `#endif`, like I see in some other places?)

Reply:
This constructor already loops over all entries of `continuity`, so maybe you can set a flag `bool validInput = true;` to `false` if one entry is not valid, and just do a single `MFEM_ASSERT` at the end of the constructor.

Reply:
I did some simplification of the `KnotVector` constructor.  I added an initial assert to avoid passing a bad `num_knots` to `Vector::SetSize()`, and then check consistency of order/continuity inside the loop.